corgidrp/mocks.py

```diff
@@ -261,6 +264,160 @@ def create_simflat_dataset(filedir=None, numfiles=10):
     dataset = data.Dataset(frames)
     return dataset
 
+
+def create_raster(mask,data,dither_sizex=None,dither_sizey=None,row_cent = None,col_cent = None,n_dith=1,mask_size=180,snr=250,planims=None):
+    """Performs raster scan of Neptune or Uranus images
+    
+     Args:
+        mask (int): (Required)  Mask used for the image.
+        data (float):(Required) Data in array npixels*npixels format to be raster scanned
+        dither_sizex (int):(Required) Size of the dither in X axis in pixels
+        dither_sizey (int):(Required) Size of the dither in X axis in pixels
+        row_cent (int): (Required)  X coordinate of the centroid
+        col_cent (int): (Required)  Y coordinate of the centroid
+        n_dith (int): number of dithers required
+        mask_size (int): Size of the mask in pixels
+        snr (int): Required SNR in the planet images
+        planims (str): Planet and band
+        
+	Returns:
+    	median dithers (np.array): median dither images
+    	mask (np.array): mask used for the dithers
+    	final (np.array): final image
+    	data_display (np.array): data 
+    	dither_stack_norm (np.array): stacked dithers
+    	full_mask (np.array) : mask used for the dithers
+    	cent (np.array): centroid of images
+    	
+        
+    """  
+ 
+    cents = []
+    
+    data_display = data.copy()
+    col_max = int(col_cent) + int(mask_size/2)
+    col_min = int(col_cent) - int(mask_size/2)
+    row_max = int(row_cent) + int(mask_size/2)
+    row_min = int(row_cent) - int(mask_size/2)
+    dithers = []
+    
+    if dither_sizey == None:
+        dither_sizey = dither_sizex
+    
+    if planims == 'neptune_band_1' or planims=='neptune_band_4':
+        planrad = 54
+        dith_end = n_dith
+    elif planims == 'uranus_band_1' or planims=='uranus_band_4':
+        dith_end = n_dith+1
+        planrad = 90
+    for i in np.arange(-n_dith,dith_end):
+        for j in np.arange(-n_dith,dith_end):
+            mask_data = data.copy()
+            image_data = mask_data[row_min + (dither_sizey * j):row_max + (dither_sizey * j), col_min + (dither_sizex * i):col_max + (dither_sizex * i)]
```

Comment:
if I'm reading this correctly, it looks like the code requires some beforehand knowledge about the dither locations to find and "lock-in" on the dither positions - one concern is that the planets are moving over the images (because Roman does not have non-sidereal tracking), so the pre-defined locations might not be an exact match to the planet's actual position - do you think your code will be robust against this? or do we need to think up an additional method?

Reply:
I don't remember exactly whether we discussed whether the dither position or some related information would be available in the header. Do we need to query the position of Uranus or Neptune somewhere in the step function?

Reply:
I would assume that we do not get any knowledge from the FITS header as to the location of Uranus/Neptune on the focal plane- I'd recommend that your code automatically finds the planet and centroids on it, to be safe

Reply:
I think it does now. This is used to create the raster-scanned images, so the information on the dither location is needed. In the `create_onsky_flatfield` step function, the planet image centroiding is done automatically, and the image is cropped in pixels depending on the planet's radius (keyword).

Reply:
ah- I missed that thanks, so this is to only *simulate* data, not to treat the real data